lib/pescarte_web/controllers/relatorio_controller.ex

```diff
@@ -0,0 +1,35 @@
+defmodule PescarteWeb.Pesquisa.RelatorioController do
+  use PescarteWeb, :controller
+
+  alias Pescarte.RelatorioCompiler
+
+  def download_pdf(conn, %{"id" => id}) do
+    pdf_binary = RelatorioCompiler.gerar_pdf(id)
+    enviar_pdf_response(conn, id, pdf_binary)
+  end
+
+  def compilar_relatorios(conn, params) do
+    campos_formularios = Map.keys(params)
+    relatorios_selecionados = Enum.filter(campos_formularios, fn id -> id != "_csrf_token" end)
+    zip_file = RelatorioCompiler.compilar_relatorios(relatorios_selecionados)
+    enviar_zip_response(conn, zip_file)
+  end
```

Comment:
creio que fique mais claro dessa forma, pois:
- você ativa diz que não quer o campo `_csrf_token`, que é único no mapa
- pega os relatorios selecionados
- compila os relatórios num zip, e devolve o nome (para não repetir uma constante) e o conteúdo do mesmo em binário
- se isso tudo der certo, dentro do `with/1/`, ai então vc manda o download numa response de sucesso

```suggestion
  def compilar_relatorios(conn, params) do
    campos_formularios = Map.delete(params, “_csrf_token")
    relatorios_selecionados = Map.keys(campos_formularios)
    
    RelatorioCompiler.compilar_relatorios(relatorios_selecionados, fn zip_filename, zip_content ->
      conn
      |> put_status(:ok)
      |> send_download({:binary, zip_content}, filename: zip_name)
    end)
  end
```